core/src/allocation.jl

```diff
@@ -952,17 +1037,22 @@ function adjust_edge_capacities!(
     p::Parameters,
     priority_idx::Int,
 )::Nothing
-    (; graph) = p
+    (; graph, allocation) = p
     (; problem, capacity, allocation_network_id) = allocation_model
+    (; main_network_connections, allocation_network_ids) = allocation
     edge_ids = graph[].edge_ids[allocation_network_id]
     constraints_capacity = problem[:capacity]
     F = problem[:F]
 
+    main_network_source_edges = get_main_network_connections(p, allocation_network_id)
+
     for edge_id in edge_ids
         c = capacity[edge_id...]
 
-        # Edges with infinite capacity have no capacity constraints
-        if isinf(c)
+        # These edges have no capacity constraints:
+        # - With infinite capacity
+        # - Being a source from the main network to a subnetwork
+        if isinf(c) || edge_id ∈ main_network_source_edges
```

Comment:
Seems like `prevfloat` would not work here.

```
julia> isinf(prevfloat(Inf))
false
```

Reply:
There's the sparse capacity matrix `AllocationModel.capacity` which is determined at initialization which was already allowed to have inifinities, which is related to but not exactly the same as the capacity values in the capacity constraints in the optimization problem. As it turns out, having infinite capacities in these constraints works fine.